tests/test_agrregation_region_aggrfuncs.py

```diff
@@ -0,0 +1,241 @@
+import pyam
+import pandas as pd
+import numpy as np
+import math
+import pytest
+
+
+all_regs = ['AFR', 'CPA', 'EEU', 'FSU', 'LAM']
+r2_northregs = ['EEU', 'FSU']
+r2_otherregs = ['AFR', 'CPA', 'LAM']
+
+WAVG_DF = pd.DataFrame([
+    ['model_a', 'scen_a', 'region_a', 'Primary Energy', 'EJ/y', 1, 6.],
+    ['model_a', 'scen_a', 'region_b', 'Primary Energy', 'EJ/y', 2, 12.],
+    ['model_a', 'scen_a', 'region_a', 'Price', 'EJ/y', 10, 5.],
+    ['model_a', 'scen_a', 'region_b', 'Price', 'EJ/y', 4, 5.],
+], columns=['model', 'scenario', 'region', 'variable', 'unit', 2005, 2010])
+
+
+@pytest.fixture(scope="function")
+def wavg_df():
+    df = pyam.IamDataFrame(data=WAVG_DF)
+    yield df
+
+
+TEST2_DF = pd.DataFrame([
+    ['MyModel', 'MyScen', 'AFR', 'FE', 'EJ/yr', 11.2, 16.8, 20.9],
+    ['MyModel', 'MyScen', 'CPA', 'FE', 'EJ/yr', 47.9, 66.7, 84.1],
+    ['MyModel', 'MyScen', 'EEU', 'FE', 'EJ/yr', 8.6, 8.6, 8.9],
+    ['MyModel', 'MyScen', 'FSU', 'FE', 'EJ/yr', 24.8, 26, 29.5],
+    ['MyModel', 'MyScen', 'LAM', 'FE', 'EJ/yr', 20.6, 24.3, 29.5],
+    ['MyModel', 'MyScen', 'AFR', 'POP', 'million', 719, 811, 1021],
+    ['MyModel', 'MyScen', 'CPA', 'POP', 'million', 1443, 1460, 1510],
+    ['MyModel', 'MyScen', 'EEU', 'POP', 'million', 126, 125, 124],
+    ['MyModel', 'MyScen', 'FSU', 'POP', 'million', 278, 279, 281],
+    ['MyModel', 'MyScen', 'LAM', 'POP', 'million', 552, 584, 643],
+    ['MyModel', 'MyScen', 'World', 'FE', 'EJ/yr', 113.1, 142.4, 172.9],
+    ['MyModel', 'MyScen', 'World', 'POP', 'million', 3118, 3259, 3579],
+], columns=['model', 'scenario', 'region', 'variable', 'unit',
+            2005, 2010, 2020])
+
+
+# for record in caplog.records:
+#    # assert record.levelname != 'CRITICAL'
+#    assert record.levelname == 'INFO'  # level == logging.level.ERROR
+# assert 'cannot aggregate variable' in caplog.text
+# assert ' because it does not exist in any subregion' in caplog.text
+# assert caplog.records[0].levelname == 'ERROR' # actually INFO
+# assert len(idf.data.columns) == 7
+# pd.testing.assert_frame_equal(TEST_DF, df.data)
+# @pytest.mark.xfail(reason="xxx")
+
+
+def test_single_region_agg(reg_df):
+    for m in ['min', 'max', 'avg', 'sum']:
+        my_df = pyam.IamDataFrame(reg_df.data)
+        my_df.aggregate_region('Primary Energy', region='A1_MEA', method=m,
+                               subregions=['MEA'], append=True)
+        np.testing.assert_allclose(
+            list(my_df.filter(region='A1_MEA').data.value),
+            list(my_df.filter(region='MEA',
+                              variable='Primary Energy').data.value))
+        assert list(my_df.filter(region='A1_MEA').data.value) == \
+            list(my_df.filter(region='MEA',
+                              variable='Primary Energy').data.value)
+
+
+def test_single_region_agg_nan_value(reg_df):
+    for m in ['min', 'max', 'avg']:
```

Comment:
Same comments as above. Please check explicitly that the function returns an `np.nan` rather than using `dropna()`, and add a docstring that this is how the functions work.

Reply:
Hm. What actually happens is that for mn/max/avg there is no "data point" returned for the year having np.nan in input. (in case of sum we get sum(Nan) = 0. for that one)